### PR TITLE
test(e2e): add preconf state consistency test

### PR DESCRIPTION
**Pull request by fridrik01**
Our existing `TestPreconfTransactions` e2e test check account state only at the end of the run, after all TXs are canonically settled. Intermediate state bugs (e.g. a flashblock exposing partial state on the pending tag) could self-heal by end-of-test and pass silently. 

This PR adds `TestFlashblockStateConsistency`. After each ETH transfer it asserts, on the preconf RPC, that the recipient balance and sender nonce are exactly as expected,  both at the pending tag (flashblock state) and at the inclusion (finalized) block. A Parallel subtest runs one worker per test account with independent recipients exercising multiple TXs per flashblock.

**Test Plan**

```
make test-e2e-preconf
```

### CODE REVIEW

File: testing/e2e/preconf/state_consistency_test.go [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	var wg sync.WaitGroup
+	for i := range senders {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			results[i] = s.runStateConsistencyWorker(
+				ctx, preconf, senders[i], newRandomAddress(),
+				stateConsistencyTxsPerWorker, gasTipCap, gasFeeCap,
+			)
+		}()
```
[the remark this answers is not in the document]
Comment (fridrik01) [reply testing/e2e/preconf/state_consistency_test.go:93]: not an issue after go 1.22

**Review by calbera — APPROVED** [review]
lgtm, noting a potential flakiness from claude:

**Pending-state atomicity:**
`waitForTxInPending` only confirms the tx appears in *some* pending snapshot. The
subsequent `PendingBalanceAt` / `PendingNonceAt` calls are separate RPC round-trips.
Between them, the sequencer may advance the flashblock or seal it into a canonical block.

- Sealing is safe: once the tx is canonical, pending state = latest canonical state +
  pending mempool, so recipient balance/sender nonce still include the tx's effect.
- The corner case that could produce false negatives is a reorg or pending-block reset on
  the preconf RPC, which should not happen on a healthy testnet.

This isn't a correctness bug in the test logic but is worth noting as a source of
potential flake. A stronger formulation would re-poll pending until the observed state
matches (with a bounded retry) rather than requiring exact match on the first read.

**Comment by fridrik01** [comment]
> lgtm, noting a potential flakiness from claude:
> 
> **Pending-state atomicity:** `waitForTxInPending` only confirms the tx appears in _some_ pending snapshot. The subsequent `PendingBalanceAt` / `PendingNonceAt` calls are separate RPC round-trips. Between them, the sequencer may advance the flashblock or seal it into a canonical block.
> 
> * Sealing is safe: once the tx is canonical, pending state = latest canonical state +
>   pending mempool, so recipient balance/sender nonce still include the tx's effect.
> * The corner case that could produce false negatives is a reorg or pending-block reset on
>   the preconf RPC, which should not happen on a healthy testnet.
> 
> This isn't a correctness bug in the test logic but is worth noting as a source of potential flake. A stronger formulation would re-poll pending until the observed state matches (with a bounded retry) rather than requiring exact match on the first read.

don't think this is an issue, we are mainly interested in finding state inconsistencies here, so re-polling would kinda defeat the point. If there are inconsistencies between flashblocks and canonical state we want the test to catch them, not retry past them.
